io/log_stream.go

```diff
@@ -0,0 +1,38 @@
+package io
+
+import (
+	"context"
+	"io"
+)
+
+type logBody struct {
+	ctx    context.Context
+	fmtstr string
+	log    func(format string, args ...interface{})
+	input  io.ReadCloser
+}
+
+func newLogBody(ctx context.Context, fmtstr string, log func(format string, args ...interface{}), rc io.ReadCloser) io.ReadCloser {
+	return &logBody{
+		ctx:    ctx,
+		fmtstr: fmtstr,
+		input:  rc,
+		log:    log,
+	}
+}
+
+func (lb *logBody) Read(p []byte) (int, error) {
+	n, err := lb.input.Read(p)
+	if n > 0 {
+		lb.log("%s%s", lb.fmtstr, p)
+	}
+	return n, err
+}
```

Comment:
This is really all it is. Why do we need a new package?

Reply:
it's not a lot of code but it makes sense with more context. It's also reusable if someone wants to plug this into their code as it just requires a logger func with a format string.
I think also little code is good. :)

Reply:
If someone needs this they can copy-poste 6 lines, no?

Reply:
What is your argument to introduce whole new exported package before we have any other use for it?

Reply:
It's actually in use @AlexanderYastrebov :)

Reply:
I am not sure its a good idea to introduce a package names as in stdlib. This will always catch an eye or need to be imported with alias like `sio` or `skpio`. Maybe we should name it like that right away.